src/sentry/search/snuba/executors.py

```diff
@@ -173,122 +288,62 @@ def snuba_search(
-                                search_filter.value.raw_value,
-                            )
-                        )
-                    )
-                else:
-                    group_categories.update(
-                        GROUP_TYPE_TO_CATEGORY[GroupType(value)]
-                        for value in search_filter.value.raw_value
-                    )
-
-            if (
-                # Don't filter on postgres fields here, they're not available
-                search_filter.key.name in self.postgres_only_fields
-                or
-                # We special case date
-                search_filter.key.name == "date"
-            ):
-                continue
-
-            converted_filter = convert_search_filter_to_snuba_query(
-                search_filter,
-                params={
-                    "organization_id": organization_id,
-                    "project_id": project_ids,
-                    "environment": environments,
-                },
-            )
-            converted_filter = self._transform_converted_filter(
-                search_filter, converted_filter, project_ids, environment_ids
-            )
-            if converted_filter is not None:
-                # Ensure that no user-generated tags that clashes with aggregation_defs is added to having
-                if search_filter.key.name in self.aggregation_defs and not search_filter.key.is_tag:
-                    having.append(converted_filter)
-                else:
-                    conditions.append(converted_filter)
-
-        extra_aggregations = self.dependency_aggregations.get(sort_field, [])
-        required_aggregations = set([sort_field, "total"] + extra_aggregations)
-        for h in having:
-            alias = h[0]
-            required_aggregations.add(alias)
-
-        aggregations = []
-        for alias in required_aggregations:
-            aggregation = self.aggregation_defs[alias]
-            if callable(aggregation):
-                # TODO: If we want to expand this pattern we should probably figure out
-                # more generic things to pass here.
-                aggregation = aggregation(start, end)
-            aggregations.append(aggregation + [alias])
-
-        if cursor is not None:
-            having.append((sort_field, ">=" if cursor.is_prev else "<=", cursor.value))
+        snuba_search_filters = [
+            sf
+            for sf in search_filters or ()
+            # remove any search_filters that are only available in postgres
+            # We special case date
+            if not (sf.key.name in self.postgres_only_fields or sf.key.name == "date")
+        ]
 
-        selected_columns = []
         if get_sample:
-            query_hash = md5(json.dumps(conditions).encode("utf-8")).hexdigest()[:8]
-            selected_columns.append(["cityHash64", [f"'{query_hash}'", "group_id"], "sample"])
-            sort_field = "sample"
```

Comment:
`sort_field` is used later on, I think we need to keep this here

Reply:
damn .. nice catch.